tools/FakeItEasy.SourceGenerators/StronglyTypedExtensionsGenerator.cs

```diff
@@ -0,0 +1,103 @@
+namespace FakeItEasy.SourceGenerators;
+
+using System;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+
+[Generator]
+public class StronglyTypedExtensionsGenerator : ISourceGenerator
+{
+    private static readonly string[] Ordinals =
+    [
+        "zeroth",
+        "first",
+        "second",
+        "third",
+        "fourth",
+        "fifth",
+        "sixth",
+        "seventh",
+        "eighth"
+    ];
+
+    public void Initialize(GeneratorInitializationContext context)
+    {
+        // No initialization required
+    }
+
+    public void Execute(GeneratorExecutionContext context)
+    {
+        GenerateArgumentValidationConfigurationExtensions(context);
+    }
+
+    private static void GenerateArgumentValidationConfigurationExtensions(GeneratorExecutionContext context)
+    {
+        const string mainClassNamespace = "FakeItEasy";
+        const string mainClassSimpleName = "ArgumentValidationConfigurationExtensions";
+        const string fullyQualifiedMainClassName = $"{mainClassNamespace}.{mainClassSimpleName}";
+        var mainClass = context.Compilation.GetTypeByMetadataName(fullyQualifiedMainClassName);
+        if (mainClass == null)
+        {
+            throw new InvalidOperationException($"Could not find the {fullyQualifiedMainClassName} class");
+        }
+
+        var source = $$"""
+                          namespace {{mainClassNamespace}}
+                          {
+                              using System;
+                              using System.Reflection;
+                              using FakeItEasy.Configuration;
+
+                              /// <content>
+                              /// Contains auto-generated functionality for the {{mainClassSimpleName}} class.
+                              /// </content>
+                              public static partial class {{mainClassSimpleName}}
+                              {
+                          {{WhenArgumentsMatchMethods(8)}}
+                              }
+                          }
+                          """;

```

Comment:
nit: reduce indentation for better readability
```suggestion
        var source = $$"""
            namespace {{mainClassNamespace}}
            {
                using System;
                using System.Reflection;
                using FakeItEasy.Configuration;

                /// <content>
                /// Contains auto-generated functionality for the {{mainClassSimpleName}} class.
                /// </content>
                public static partial class {{mainClassSimpleName}}
                {
            {{WhenArgumentsMatchMethods(8)}}
                }
            }
            """;
```

Reply:
Ah, but if we did that, the output would look funny. Too outdented. If you'd still rather it look nice, and I'm too worked up about ugly generated code, I could post-process the lines to indent.

Reply:
Are you sure? I think the closing `"""` determines the output's indentation level.